dot/build_spec_test.go

```diff
@@ -0,0 +1,289 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package dot
+
+import (
+	"errors"
+	"os"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/lib/genesis"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBuildSpec_ToJSON(t *testing.T) {
+	type fields struct {
+		genesis *genesis.Genesis
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   string
+		err    error
+	}{
+		{
+			name:   "name test",
+			fields: fields{genesis: &genesis.Genesis{Name: "test"}},
+			want: `{
+    "name": "test",
+    "id": "",
+    "chainType": "",
+    "bootNodes": null,
+    "telemetryEndpoints": null,
+    "protocolId": "",
+    "genesis": {},
+    "properties": null,
+    "forkBlocks": null,
+    "badBlocks": null,
+    "consensusEngine": "",
+    "codeSubstitutes": null
+}`,
+		},
+		{
+			name: "additional parameters test",
+			fields: fields{genesis: &genesis.Genesis{
+				Name:            "test",
+				ID:              "ID",
+				ChainType:       "chainType",
+				ProtocolID:      "protocol",
+				ConsensusEngine: "babe",
+			}},
+			want: `{
+    "name": "test",
+    "id": "ID",
+    "chainType": "chainType",
+    "bootNodes": null,
+    "telemetryEndpoints": null,
+    "protocolId": "protocol",
+    "genesis": {},
+    "properties": null,
+    "forkBlocks": null,
+    "badBlocks": null,
+    "consensusEngine": "",
+    "codeSubstitutes": null
+}`,
+		},
+		{
+			name: "normal conditions",
+			fields: fields{genesis: &genesis.Genesis{
+				Name:               "test",
+				ID:                 "ID",
+				ChainType:          "chainType",
+				Bootnodes:          []string{"node1", "node2"},
+				TelemetryEndpoints: []interface{}{"endpoint"},
+				ProtocolID:         "protocol",
+				Genesis:            genesis.Fields{},
+				Properties:         map[string]interface{}{"key": "value"},
+				ForkBlocks:         []string{"1", "2"},
+				BadBlocks:          []string{"3", "4"},
+				ConsensusEngine:    "babe",
+				CodeSubstitutes:    map[string]string{"key": "value"},
+			}},
+			want: `{
+    "name": "test",
+    "id": "ID",
+    "chainType": "chainType",
+    "bootNodes": [
+        "node1",
+        "node2"
+    ],
+    "telemetryEndpoints": null,
+    "protocolId": "protocol",
+    "genesis": {},
+    "properties": {
+        "key": "value"
+    },
+    "forkBlocks": null,
+    "badBlocks": null,
+    "consensusEngine": "",
+    "codeSubstitutes": null
+}`,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			b := &BuildSpec{
+				genesis: tt.fields.genesis,
+			}
+			got, err := b.ToJSON()
+			if tt.err != nil {
+				assert.ErrorIs(t, err, tt.err)
+			} else {
+				assert.NoError(t, err)
+			}
```

Comment:
This makes sense, I've got a few places I used this if/else check, I'll replace with assert.ErrorIs

Reply:
Nice 👍 It also depends, if you use `assert.EqualError(t, err, errMessage)` then you need the nil check since a nil error would result in the string `<nil>` and having your test case `errMessage` set as `<nil>` is just strange/cumbersome.

Reply:
Note: `assert.ErrorIs` may not work for all checks due to some errors causing "Target error should be in err chain:" errors.